core/trino-spi/src/main/java/io/trino/spi/ThreadsCount.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi;
+
+import java.util.function.Supplier;
+
+import static java.lang.Math.round;
+
+public record ThreadsCount(int threadsCount)
+        implements Comparable<ThreadsCount>
+{
+    public ThreadsCount
+    {
+        if (threadsCount < 0) {
+            throw new IllegalArgumentException("Threads number cannot be smaller than 0");
+        }
+    }
+
+    private static final String PER_CORE_SUFFIX = "C";
+    private static final String PER_CORE_NEXT_POWER_OF_2_SUFFIX = "C2";
```

Comment:
It's confusing as it suggest it's `2` number of cores rather than pow of two. See how we deal with it in `io.trino.execution.TaskManagerConfig#taskConcurrency`.